backend/fms_core/template_importer/row_handlers/normalization/normalization.py

```diff
@@ -0,0 +1,92 @@
+
+from fms_core.models import ProcessMeasurement
+
+from fms_core.services.container import get_container, get_or_create_container
+from fms_core.services.sample import get_sample_from_container, transfer_sample, update_sample
+from fms_core.services.property_value import create_process_measurement_properties
+
+from fms_core.utils import convert_concentration_from_nm_to_ngbyul
+
+
+class NormalizationRowHandler(GenericRowHandler):
+
+    def process_row_inner(self, source_sample, destination_sample, process_measurement, process_measurement_properties):
+        # Check case when both concentrations are given or none are given
+        if destination_sample['concentration_nm'] is None and destination_sample['concentration_ngul'] is None:
+            self.errors['concentration'] = 'A concentration in either nM or ng/uL must be specified.'
+
+        if destination_sample['concentration_nm'] is not None and destination_sample['concentration_ngul'] is not None:
+            self.errors['concentration'] = 'Concentration must be specified in either nM or ng/uL, not both.'
+
+        source_sample_obj, self.errors['sample'], self.warnings['sample'] = get_sample_from_container(
+            barcode=source_sample['container']['barcode'],
+            coordinates=source_sample['coordinates'])
+
+        destination_container_dict = destination_sample['container']
+
+        parent_barcode = destination_container_dict['parent_barcode']
+        if parent_barcode:
+            container_parent_obj, self.errors['parent_container'], self.warnings['parent_container'] = get_container(
+                barcode=parent_barcode)
+        else:
+            container_parent_obj = None
+
+        if source_sample_obj and (container_parent_obj or not parent_barcode) and "concentration" not in self.errors.keys():
+            destination_container, _, self.errors['container'], self.warnings['container'] = get_or_create_container(
+                barcode=destination_container_dict['barcode'],
+                kind=destination_container_dict['kind'],
+                name=destination_container_dict['name'],
+                coordinates=destination_container_dict['coordinates'],
+                container_parent=container_parent_obj)
+
+            resulting_sample, self.errors['transfered_sample'], self.warnings['transfered_sample'] = transfer_sample(
+                process=process_measurement['process'],
+                sample_source=source_sample_obj,
+                container_destination=destination_container,
+                volume_used=process_measurement['volume_used'],
+                execution_date=process_measurement['execution_date'],
+                coordinates_destination=destination_sample['coordinates'],
+                volume_destination=destination_sample['volume'],
+                source_depleted=source_sample['depleted'],
+                comment=process_measurement['comment'])
+
+            concentration = None
```

Comment:
Move the whole concentration calculation before the container creation and  before moving forward make a validation that the volume used, source_sample concentration, final volume and final concentration add up. It will likely not be an exact match but you can round both values up to 1 decimal (when comparing theoretical to submitted concentration) and make sure it match at this level...